src/IB/IIMethod.cpp

```diff
@@ -0,0 +1,4444 @@
+const std::string IIMethod::TAU_IN_SYSTEM_NAME = "Interior traction system";
+const std::string IIMethod::TAU_OUT_SYSTEM_NAME = "Exterior traction system";
+const std::array<std::string, NDIM> IIMethod::VELOCITY_JUMP_SYSTEM_NAME = { { "velocity [[du]] jump system",
+                                                                              "velocity [[dv]] jump system"
+#if (NDIM == 3)
+                                                                              ,
+                                                                              "velocity [[dw]] jump system"
+#endif
+} };
+
+/////////////////////////////// PUBLIC ///////////////////////////////////////
+
+IIMethod::IIMethod(const std::string& object_name,
+                   Pointer<Database> input_db,
+                   MeshBase* mesh,
+                   int max_level_number,
+                   bool register_for_restart,
+                   const std::string& restart_read_dirname,
+                   unsigned int restart_restore_number)
+{
+    commonConstructor(object_name,
+                      input_db,
+                      std::vector<MeshBase*>(1, mesh),
+                      max_level_number,
+                      register_for_restart,
+                      restart_read_dirname,
+                      restart_restore_number);
+    return;
+} // IIMethod
+
+IIMethod::IIMethod(const std::string& object_name,
+                   Pointer<Database> input_db,
+                   const std::vector<MeshBase*>& meshes,
+                   int max_level_number,
+                   bool register_for_restart,
+                   const std::string& restart_read_dirname,
+                   unsigned int restart_restore_number)
+    : d_num_parts(static_cast<int>(meshes.size()))
+{
+    commonConstructor(object_name,
+                      input_db,
+                      meshes,
+                      max_level_number,
+                      register_for_restart,
+                      restart_read_dirname,
+                      restart_restore_number);
+    return;
+} // IIMethod
+
+IIMethod::~IIMethod()
+{
+    for (unsigned int part = 0; part < d_num_parts; ++part)
+    {
+        delete d_equation_systems[part];
+    }
+    if (d_registered_for_restart)
+    {
+        RestartManager::getManager()->unregisterRestartItem(d_object_name);
+        d_registered_for_restart = false;
+    }
+    return;
+} // ~IIMethod
+
+FEDataManager*
+IIMethod::getFEDataManager(const unsigned int part) const
+{
+    TBOX_ASSERT(d_fe_equation_systems_initialized);
+    TBOX_ASSERT(part < d_num_parts);
+    return d_fe_data_managers[part];
+} // getFEDataManager
+
+void
+IIMethod::registerDisconElemFamilyForJumps(const unsigned int part)
+{
+    TBOX_ASSERT(!d_fe_equation_systems_initialized);
+    TBOX_ASSERT(part < d_num_parts);
+    d_use_discon_elem_for_jumps[part] = true;
+    return;
+} // registerTangentialVelocityMotion
```

Comment:
```suggestion
} // registerDisconElemFamilyForJumps
```

Reply:
Actually, if we are updating this anyway, I'd suggest `setJumpFEFamily` and then pass in the FE family.

Also, I think we usually set up things like `FEFamily` choices in the input file. I am not sure that we can switch the `FEFamily` after everything has been initialized. I wonder if it is better to only allow this to be done in the input file?

Reply:
Okay, but what if this is only an attribute for certain part(s)? I think for smooth geometries we would prefer to continue using continuous Lagrange family. This function allows for registering the discontinuous FE family for a certain part in the main file.

Reply:
Fair enough. Is there logic to confirm that the settings are being changed prior to initializing everything?

Does it make sense to allow users to pass in the `FEFamily`?

I guess the alternative is to let users specify the continuous and discontinuous `FEFamily` choices in the input file, and then switch between them for a given part. Suggest something like `setJumpConditionFEFamily{Continuous,Discontinuous}`.

Reply:
So currently there is an assertion to confirm that FE equation systems are not initialized prior to registering with parts. Additionally, the user passes the following in the input file which only gets activated if the registration is also done with the part in the main file, otherwise everything is done using regular Lagrange family:
```
   pressure_jump_fe_family = "L2_LAGRANGE"
   velocity_jump_fe_family = "L2_LAGRANGE"
   wss_fe_family = "L2_LAGRANGE"
   tau_fe_family = "L2_LAGRANGE"
```
I realize though that this can actually be done more neatly all in `registerDisconElemFamilyForJumps` maybe

Reply:
These are fixes that are worth doing but they are beyond the scope of this PR - could you open a separate issue for this so we can follow up later?

Reply:
done.